packages/iota/src/highLevel/addressBalance.ts

```diff
@@ -53,6 +54,21 @@ export async function addressBalance(
         cursor = response.cursor;
     } while (cursor && response.items.length > 0);
 
+    do {
+        expirationResponse = await indexerPluginClient.outputs({
+            expirationReturnAddressBech32: addressBech32, expiresBefore: Math.floor(Date.now() / 1000), cursor
```

Comment:
Reusing the same `cursor` without "resetting" ? This could result in first request having a `cursor` from `do` block above which is unrelated.